doc/source/dev/missing-bits.rst

```diff
@@ -87,6 +87,19 @@ private return classes, please see  how `~scipy.stats.binomtest` summarizes
 `~scipy.stats._result_classes.BinomTestResult` and links to its documentation,
 and note that ``BinomTestResult`` cannot be imported from `~scipy.stats`.
 
+Depending on the complexity of "MyResultObject", a normal class or a dataclass
+can be used. When using dataclasses, do not use ``dataclasses.make_dataclass``,
```

Comment:
Just in case you want to add that reference:
```suggestion
can be used. When using dataclasses, do not use :py:func:`dataclasses.make_dataclass`,
```